### ISSUE TITLE
Potential Bugs and Questions on Ray Direction and Camera-to-World Transformation in water_splatting.py

### ISSUE THREAD

**Issue by Jack-Feng1672** [water-splatting/water-splatting#7]
Dear Author,

I have encountered **two potential issues** related to **ray direction calculations** and the **camera2world transformation** in the code, specifically in the `water_splatting.py` file between lines [**732-761**](https://github.com/water-splatting/water-splatting/blob/main/water_splatting/water_splatting.py#L732-L761
). The relevant code section is as follows:

```python
R = camera.camera_to_worlds[0, :3, :3]  # 3 x 3
T = camera.camera_to_worlds[0, :3, 3:4]  # 3 x 1
# flip the z and y axes to align with gsplat conventions
R_edit = torch.diag(torch.tensor([1, -1, -1], device=self.device, dtype=R.dtype))
R = R @ R_edit
# analytic matrix inverse to get world2camera matrix
R_inv = R.T
T_inv = -R_inv @ T

...

# Medium
# Encode directions
y = torch.linspace(0., H, H, device=self.device)
x = torch.linspace(0., W, W, device=self.device)
yy, xx = torch.meshgrid(y, x)
yy = (yy - cy) / camera.fy.item()
xx = (xx - cx) / camera.fx.item()
directions = torch.stack([yy, xx, -1 * torch.ones_like(xx)], dim=-1)
norms = torch.linalg.norm(directions, dim=-1, keepdim=True)
directions = directions / norms
directions = directions @ R
``` 

## Issue 1: Ray Direction Calculation
I noticed that you converted the camera coordinate system from the OpenGL convention to the COLMAP convention through the `R = R @ R_edit`. In the OpenCV (COLMAP) convention, in the camera coordinate system, the image plane's x-axis is to the right, the y-axis is downward, and the z-axis is pointing forward. Therefore, the direction for each pixel in the camera coordinate system should be:

```python
directions = torch.stack([xx, yy, torch.ones_like(xx)], dim=-1)
``` 
**However**, in the code, the directions are calculated as:
```python
directions = torch.stack([yy, xx, -1 * torch.ones_like(xx)], dim=-1)
``` 
**This does not seem to follow the COLMAP convention.** This calculation does not align with the COLMAP convention, which has caused some confusion. Could you kindly confirm if this calculation is intentional or if it requires adjustment to reflect the correct convention?

## Issue 2: Camera to World Transformation
After calculating the normalized directions in the camera coordinate system, the directions are transformed from camera coordinates to the world coordinates. According to the standard camera transformation formula, the direction of a ray in world coordinates should be calculated as follows:

Given a ray direction in the camera coordinate system $$\boldsymbol{P}_c=[x,y,z]$$ (a row vector), and a rotation matrix $\boldsymbol{R}$ (from camera to world coordinates), the ray direction  $$\boldsymbol{P}_w$$  in world coordinates should be: 

$$ \boldsymbol{P}_w^T = \boldsymbol{R} @ \boldsymbol{P}_c^T$$ 

or equivalently,

$$ \boldsymbol{P}_w = \boldsymbol{P}_c @ \boldsymbol{R}^T$$ 

Since in the PyTorch implementation, the tensor `directions` has the shape $$[H, W, 3]$$, it suggests that the directions in world coordinates should be:
```python
directions = directions @ R.T
``` 
or calculated according to [SeathruNeRF-NS](https://github.com/nerfstudio-project/nerfstudio/blob/main/nerfstudio/cameras/cameras.py#L892-L895)

```python
directions = torch.sum(directions.unsqueeze(-2) * R, -1)
```

**However** in the code, it is instead computed as:
```python
directions = directions @ R
``` 
**This inconsistency has caused some confusion.** Could you please clarify if this calculation is correct or if it needs to be revised to align with the standard transformation formula?

Thank you for your time and help in clarifying these points. I look forward to your response.

Best regards.

**Comment by cgwu1999** [comment]
I also found this problem, but when I fixed these two problems, the model's performance went down instead
![Image](https://github.com/user-attachments/assets/62248765-ae45-4615-8087-2efcb9813d2c)
